test-suite-data.json

```
@@ -466,5 +466,65 @@
     "qualifiers": null,
     "subpath": null,
     "is_invalid": true
+  },
+  {
+    "description": "a name and version are always required",
+    "purl": "pkg:pub",
+    "canonical_purl": "pkg:pub",
+    "type": "pub",
+    "namespace": null,
+    "name": null,
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": true
+  },
+  {
+    "description": "valid base pub purl",
+    "purl": "pkg:pub",
+    "canonical_purl": "pkg:pub/utils@0.1.0",
```

```suggestion
    "description": "valid base pub purl",
    "purl": "pkg:pub/utils@0.1.0",
    "canonical_purl": "pkg:pub/utils@0.1.0",
```

`purl` should be a valid PURL that parsers canonicalize to obtain `canonical_purl`.